api/ausy_encoder.c

```diff
@@ -0,0 +1,128 @@
+#include "ausy_encoder.h"
+#include "deps/json-parser/json.h"
+#include "oc_log.h"
+#include "deps/tinycbor/src/cborjson.h"
+
+static CborError ausy_encode_value_2_cbor(json_value* value, CborEncoder* currentEncoder);
+
+static CborError ausy_encode_map_2_cbor(json_value* value, CborEncoder* currentEncoder)
+{
+    CborError err = CborNoError;
+    int length, x;
+    if (value == NULL) {
+        return err;
+    }
+    length = value->u.object.length;
+    for (x = 0; x < length; x++) {
+        const char* key = value->u.object.values[x].name;
+        err |= cbor_encode_text_stringz(currentEncoder, key);
```

Comment:
I would break once the first error is encountered and print error (`OC_ERR`) with `key` so we know which attribute caused the problem